WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift

```diff
@@ -150,18 +156,46 @@ class AztecPostViewController: UIViewController {
 
     /// Post being currently edited
     ///
-    fileprivate(set) var post: AbstractPost {
+    dynamic fileprivate(set) var post: AbstractPost {
         didSet {
+            removeObservers(fromPost: oldValue)
+            addObservers(toPost: post)
+
             refreshInterface()
         }
     }
 
 
+    /// Maintainer of state for editor - like for post button
+    ///
+    fileprivate(set) lazy var postEditorStateContext: PostEditorStateContext = {
+        let originalPostStatus: PostStatus
+
+        if let originalPost = self.post.original,
+            let postStatus = originalPost.status,
+            originalPost.hasRemote() {
+            originalPostStatus = PostStatus(rawValue: postStatus) ?? .draft
+        } else {
+            originalPostStatus = .draft
+        }
```

Comment:
Nipticky // feel free to disregard!!

Potential candidate for a helper method:

```
enum PostStatus {
   static func fromPost(post: Post) -> PostStatus {
 //...
   }
}
```